core/test/utils/assertions.hpp

```diff
@@ -202,6 +202,86 @@ void print_columns(Ostream& os, const Iterator& begin, const Iterator& end)
 }
 
 
+template <typename Ostream, typename MatrixData1, typename MatrixData2>
+void print_sparsity_pattern(Ostream& os, const MatrixData1& first,
+                            const MatrixData2& second)
+{
+    auto first_it = first.nonzeros.begin();
+    auto second_it = second.nonzeros.begin();
+    os << ' ';
+    for (int col = 0; col < first.size[1]; col++) {
+        os << (col % 10);
+    }
+    os << '\n';
+    for (int row = 0; row < first.size[0]; row++) {
+        os << (row % 10);
+        for (int col = 0; col < first.size[1]; col++) {
+            const auto has_first =
+                get_next_value(first_it, end(first.nonzeros), row, col) !=
+                gko::zero<typename MatrixData1::value_type>();
+            const auto has_second =
+                get_next_value(second_it, end(second.nonzeros), row, col) !=
+                gko::zero<typename MatrixData2::value_type>();
+            if (has_first) {
+                if (has_second) {
+                    os << '+';
+                } else {
+                    os << '|';
+                }
+            } else {
+                if (has_second) {
+                    os << '-';
+                } else {
+                    os << ' ';
+                }
+            }
+        }
+        os << (row % 10) << '\n';
+    }
+    os << ' ';
+    for (int col = 0; col < first.size[1]; col++) {
+        os << (col % 10);
+    }
+    os << "\n| is first, - is second, + is both\n";
```

Comment:
Maybe mention that space is neither?
```suggestion
    os << "\n'|' is first, '-' is second, '+' is both, ' ' is neither\n";
```